File: backend/src/main/java/com/richstone/cargo/dto/QuestionDto.java
package com.richstone.cargo.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class QuestionDto {
    @Schema(description = "Текст вопроса", example = "What is the largest planet?")
    private String question;

    @Schema(description = "Ответ на вопрос", example = "Jupiter")
    private String answer;
}

File: android/app/src/main/java/com/example/richstonecargo/data/remote/dto/QuestionDto.kt
package com.example.richstonecargo.data.remote.dto

import com.example.richstonecargo.domain.model.Question

data class QuestionDto(
    val id: Long?,
    val question: String?,
    val answer: String?
)

fun QuestionDto.toQuestion(): Question {
    return Question(
        id ?: 0,
        question ?: "",
        answer ?: "",
    )
}